pkg/ast/spl/spl.peg

```diff
@@ -3277,10 +3416,29 @@ EarliestAndLatest <- CMD_EARLIEST EQUAL earliestTime:AbsoluteTimestamp SPACE CMD
     return node, nil
 }
 
-EarliestOnly <- CMD_EARLIEST EQUAL earliestTime:AbsoluteTimestamp !. {
-    startEpoch, err := ConvertCustomDateTimeFormatToEpochMs(earliestTime.(string))
-    if err != nil {
-        return nil, err
+EarliestOnly <- CMD_EARLIEST EQUAL earliestTime:(AbsoluteTimestamp / RelativeTimestamp) !. {
+    var startEpoch int64 = 0
+    var err error
+
+    earliestTimeModifier := earliestTime.(ast.RelativeTimeModifier)
+
+    if earliestTimeModifier.AbsoluteTime != "" {
+        startEpoch, err = ConvertCustomDateTimeFormatToEpochMs(earliestTimeModifier.AbsoluteTime)
+        if err != nil {
+            return nil, err
+        }
+    } else {
+        t := time.Now()
+        offsetNum := earliestTimeModifier.RelativeTimeOffset.Offset
```

Comment:
I would suggest to refactor this part of the else code block into a separate function and call it wherever it is required. As I can see the similar else block is there in three places.

Reply:
From the below Parsing definition at #L3500:
`RelativeTimestamp <- offset:(Offset)? snapParam:(Snap)? { ... }`
I believe both of them can be set. 

If that is not the case, then you need to change the parser to reflect that.